torchvision/models/regnet.py

```diff
@@ -0,0 +1,708 @@
+# Modified from
+# https://github.com/facebookresearch/ClassyVision/blob/main/classy_vision/models/anynet.py
+# https://github.com/facebookresearch/ClassyVision/blob/main/classy_vision/models/regnet.py
+
+
+import numpy as np
+import math
+import torch
+
+from collections import OrderedDict
+from functools import partial
+from typing import Any, Callable, List, Optional, Tuple
+from torch import nn, Tensor
+
+from .._internally_replaced_utils import load_state_dict_from_url
+from torchvision.models.mobilenetv2 import ConvBNActivation, _make_divisible
+
+model_urls = {
+    # TODO(kazhang): add pretrained weights
+    "regnet_y_400m": "",
+}
+
+
+class _SqueezeExcitation(nn.Module):
+    """
+    Squeeze and excitation layer from
+    `"Squeeze-and-Excitation Networks" <https://arxiv.org/pdf/1709.01507>`_.
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        reduction_ratio: Optional[int] = 16,
+        reduced_channels: Optional[int] = None,
+        activation: Optional[nn.Module] = None,
+    ) -> None:
+        super().__init__()
+        self.avgpool = nn.AdaptiveAvgPool2d((1, 1))
+
+        # Either reduction_ratio is defined, or out_channels is defined,
+        # neither both nor none of them
+        assert bool(reduction_ratio) != bool(reduced_channels)
+
+        if activation is None:
+            activation = nn.ReLU()
+
+        reduced_channels = (
+            in_channels // reduction_ratio if reduced_channels is None else reduced_channels
+        )
+        self.excitation = nn.Sequential(
+            nn.Conv2d(in_channels, reduced_channels, kernel_size=1, stride=1, bias=True),
+            activation,
+            nn.Conv2d(reduced_channels, in_channels, kernel_size=1, stride=1, bias=True),
+            nn.Sigmoid(),
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        x_squeezed = self.avgpool(x)
+        x_excited = self.excitation(x_squeezed)
+        x_scaled = x * x_excited
+        return x_scaled
+
+
+class BasicTransform(nn.Sequential):
+    """Basic transformation: [3x3 conv, BN, Relu] x2."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        stride: int,
+        norm_layer: Callable[..., nn.Module],
+        activation_layer: Callable[..., nn.Module],
+    ) -> None:
+        super().__init__(OrderedDict(
+            a=nn.Sequential(
+                ConvBNActivation(width_in, width_out, kernel_size=3, stride=stride,
+                                 norm_layer=norm_layer, activation_layer=activation_layer),
+                nn.Conv2d(width_out, width_out, 3, stride=1, padding=1, bias=False),
+            ),
+            final_bn=norm_layer(width_out),
+        ))
+
+
+class ResStemIN(nn.Sequential):
```

Comment:
@mannatsingh could you please comment on whether we need `ResStemIN`, `VanillaBlock` and `ResBottleneckLinearBlock`? they're not currently used in predefined model architectures.

Reply:
Yes, these blocks aren't currently used for any RegNet version! See my high level comment for this :)

Reply:
I agree with the recent https://github.com/pytorch/vision/pull/4403#issuecomment-924434522 by @kazhang that any class not used should be removed. We can always add them back later if we add more models.